test/e2e/shoot/logging/event_logging.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/gardener/gardener/test/e2e"
+	"github.com/gardener/gardener/test/framework"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	eventLogger            = "event-logger"
+	eventMessageForSeed    = "Event from Event Logger integration test related to the Seed"
+	eventMessageForShoot   = "Event from Event Logger integration test related to the Shoot"
+	seedEventName          = "seed-event"
+	shootEventName         = "shoot-event"
+	eventLoggerName        = "event-logger"
+	kubeSystemNamespace    = "kube-system"
+	operatorID             = "operator"
+	lokiName               = "loki"
+	gardenNamesapce        = "garden"
+	fluentBitDaemonSetName = "fluent-bit"
+)
+
+var _ = Describe("Seed logging testing", Label("Shoot", "logging"), func() {
+
+	var (
+		shootPurpose   = gardencorev1beta1.ShootPurposeDevelopment
+		shootNamespace string
+		lokiLabels     = map[string]string{
+			"app":  "loki",
+			"role": "logging",
+		}
+	)
+
+	f := defaultShootCreationFramework()
+	f.Shoot = e2e.DefaultShoot("event-")
+	f.Shoot.Spec.Purpose = &shootPurpose
+
+	BeforeEach(func() {
+		parentCtx = context.Background()
+
+		ctx, cancel := context.WithTimeout(parentCtx, 15*time.Minute)
+		defer cancel()
+		Expect(f.CreateShootAndWaitForCreation(ctx, false)).To(Succeed())
+		f.Verify()
+
+		shootNamespace = f.Shoot.Status.TechnicalID
+
+		Eventually(func(g Gomega) {
+			g.Expect(checkRequiredEventLoggerResources(ctx, f.ShootFramework.SeedClient, shootNamespace)).To(Succeed())
+		}).Should(Succeed())
+	})
+
+	It("should get events from loki", Label("event-logger"), func() {
+		ctx, cancel := context.WithTimeout(parentCtx, 15*time.Minute)
+		defer cancel()
+
+		By("Wait until Loki StatefulSet is ready")
+		Eventually(func(g Gomega) {
+			g.Expect(f.WaitUntilStatefulSetIsRunning(ctx, lokiName, shootNamespace, f.ShootFramework.SeedClient)).To(Succeed())
+		}).Should(Succeed())
+
+		By("Wait until Event-Logger Deployment is ready")
+		Eventually(func(g Gomega) {
+			g.Expect(f.WaitUntilDeploymentIsReady(ctx, eventLogger, shootNamespace, f.ShootFramework.SeedClient)).To(Succeed())
+		}).Should(Succeed())
+
+		By("Create seed event")
+		seedEvent := getEventFor("seed", shootNamespace)
+		Expect(f.ShootFramework.SeedClient.Client().Create(ctx, seedEvent)).To(Succeed())
+
+		By("Create shoot event")
+		shootEvent := getEventFor("shoot", kubeSystemNamespace)
+		Expect(f.ShootFramework.ShootClient.Client().Create(ctx, shootEvent)).To(Succeed())
+
+		By("Wait until seed event is created")
+		framework.ExpectNoError(waitUntilEventIsCreated(ctx, f.Logger, seedEventName, shootNamespace, f.ShootFramework.SeedClient))
```

Comment:
Use `Expect(...).To(Succeed())` instead of `framework.ExpectNoError` to minimize confusion.